File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/network/http/HttpPathOverrideRequest.java
package io.embrace.android.embracesdk.internal.network.http;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.embrace.android.embracesdk.annotation.InternalApi;

@InternalApi
public interface HttpPathOverrideRequest {

    @Nullable
    String getHeaderByName(@NonNull String name);

    @NonNull
    @SuppressWarnings("AbbreviationAsWordInNameCheck")
    String getOverriddenURL(@NonNull String pathOverride);

    @NonNull
    @SuppressWarnings("AbbreviationAsWordInNameCheck")
    String getURLString();
}

File: embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/network/logging/HttpPathOverrideRequest.kt
package io.embrace.android.embracesdk.internal.network.logging

import io.embrace.android.embracesdk.annotation.InternalApi

@InternalApi
interface HttpPathOverrideRequest {
    fun getHeaderByName(name: String): String?

    fun getOverriddenURL(pathOverride: String): String

    fun getURLString(): String
}
